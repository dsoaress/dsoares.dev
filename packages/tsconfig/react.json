{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "React",
  "extends": "./base.json",
  "compilerOptions": {
    "jsx": "react-jsx",
    "lib": ["ES2015"],
    "module": "ESNext",
    "target": "es6"
  }
}
