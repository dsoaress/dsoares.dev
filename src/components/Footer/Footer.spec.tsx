import { render, screen } from '@testing-library/react'

import { Footer } from '.'

const footer = 'Footer'

jest.mock('@/components/Social', () => ({
  Social: () => <div />
}))

jest.mock('@/hooks/useData', () => ({
  useData: () => ({ t: { footer } })
}))

describe('<Footer />', () => {
  beforeEach(() => {
    render(<Footer />)
  })

  it('should matches snapshot', () => {
    expect(screen.getByTestId('footer')).toMatchSnapshot()
  })

  it('should render correct footer text', () => {
    expect(screen.getByText(footer)).toBeInTheDocument()
  })
})
