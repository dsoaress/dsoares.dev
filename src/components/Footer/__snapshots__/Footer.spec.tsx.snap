// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should matches snapshot 1`] = `
<footer
  class="my-12 flex flex-col justify-center"
  data-testid="footer"
>
  <div />
  <p
    class="font-normal text-neutral-900 transition-colors duration-300 dark:text-neutral-50 mt-6 text-center text-neutral-600 dark:text-neutral-400 text-sm leading-5"
  >
    Footer
  </p>
</footer>
`;
